### PR TITLE
pkg: allow setting rand's seed in stringutil

### PR DESCRIPTION
**Pull request by fanminshi**

### CODE REVIEW

**Comment by xiang90** [comment]
lgtm

**Comment by heyitsanthony** [comment]
I think this is a bad idea-- why can't the importer seed rand?

**Comment by xiang90** [comment]
> I think this is a bad idea-- why can't the importer seed rand?

good point. at least, we probably should use a private rand rather than the global one if we want to seed in this pkg.

**Comment by fanminshi** [comment]
@heyitsanthony If the importer sets seed in its package scope, will the seed be reflected in calling `UniqueStrings()`?

**Comment by heyitsanthony** [comment]
@fanminshi yes. It updates a global: https://golang.org/src/math/rand/rand.go?s=2322:2353#L242

**Comment by fanminshi** [comment]
I'll go with the private rand seed approach; lesser side affect than the global one.

**Comment by fanminshi** [comment]
All fixed. PTAL

File: pkg/stringutil/stringutil.go [hunk 1]
```diff
@@ -20,7 +20,18 @@ const (
...
 
-// UniqueStrings retruns a slice of randomly generated unique strings.
+var random *rand.Rand
+
+func init() {
+	random = rand.New(rand.NewSource(1))
+}
+
+// SetSeed sets the seed of stringutil's private rand
+func SetSeed(seed int64) {
```
Comment (heyitsanthony) [inline pkg/stringutil/stringutil.go]: `SetRand(r *rand.Rand)`? gives more control over the random generator

File: pkg/stringutil/stringutil.go [hunk 2]
```diff
@@ -20,7 +20,18 @@ const (
...
 )
 
-// UniqueStrings retruns a slice of randomly generated unique strings.
+var random *rand.Rand
+
+func init() {
+	random = rand.New(rand.NewSource(1))
+}
+
+// SetSeed sets the seed of stringutil's private rand
```
Comment (heyitsanthony) [inline pkg/stringutil/stringutil.go]: s/private rand/package-level random number generator.

File: pkg/stringutil/stringutil.go [hunk 3]
```diff
@@ -20,7 +20,18 @@ const (
 	chars = "abcdefghijklmnopqrstuvwxyz0123456789"
 )
 
-// UniqueStrings retruns a slice of randomly generated unique strings.
+var random *rand.Rand
```
Comment (heyitsanthony) [inline pkg/stringutil/stringutil.go]: s/random/rng for RandomNumberGenerator

File: pkg/stringutil/stringutil.go [hunk 4]
```diff
@@ -20,7 +20,18 @@ const (
...
+func init() {
+	random = rand.New(rand.NewSource(1))
+}
+
+// SetSeed sets the seed of stringutil's private rand
+func SetSeed(seed int64) {
+	random.Seed(seed)
+}
+
+// UniqueStrings retruns a slice of randomly generated unique strings
```
Comment (heyitsanthony) [inline pkg/stringutil/stringutil.go]: s/strings/strings.

**Comment by fanminshi** [comment]
discussed with @heyitsanthony, seeding at global will be sufficient for the runner for now. no need to change stringutil.

**Pull request closed without merging**